packages/select/src/ux-select.ts

```diff
@@ -1,586 +1,599 @@
+
+import { UxSelectTheme } from './ux-select-theme';
+import { UxOptGroupElement } from './ux-optgroup';
+import { UxOptionElement } from './ux-option';
+import { getAuViewModel, bool } from './util';
+
+// tslint:disable-next-line: no-submodule-imports
+import '@aurelia-ux/core/components/ux-input-component.css';
+// tslint:disable-next-line: no-submodule-imports
+import '@aurelia-ux/core/components/ux-input-component--outline.css';
+import { UxDefaultSelectConfiguration } from './ux-default-select-configuration';
+
+declare module './ux-option' {
+  interface UxOption {
+    uxSelect: UxSelect;
+  }
+}
+
+declare module './ux-optgroup' {
+  interface UxOptGroup {
+    uxSelect: UxSelect;
+  }
+}
+
+const UP = 38;
+// const RIGHT = 39;
+const DOWN = 40;
+// const LEFT = 37;
+// const ESC = 27;
+const ENTER = 13;
+const SPACE = 32;
+const logger = getLogger('ux-select');
+const invalidMultipleValueMsg = 'Only null or Array instances can be bound to a multi-select';
+const selectArrayContext = 'context:ux-select';
+
+export interface UxSelectElement<T = any> extends HTMLElement {
+  matcher(a: any, b: any): boolean;
+  value: T;
+}
+
+export interface UxOptionContainer extends HTMLElement {
+  children: HTMLCollectionOf<UxOptGroupElement | UxOptionElement>;
+}
+
+export type InputVariant = 'filled' | 'outline';
+
+@inject(Element, StyleEngine, ObserverLocator, TaskQueue, UxDefaultSelectConfiguration)
+@processContent(ensureUxOptionOrUxOptGroup)
+@customElement('ux-select')
+@useView(PLATFORM.moduleName('./ux-select.html'))
+export class UxSelect implements UxInputComponent {
+
+  private selectedOption: UxOptionElement | null = null;
+
+  // Temporarily used to store <ux-option/> reference in interaction
+  private focusedUxOption: UxOptionElement | null;
+
+  // Observe window events
+  private winEvents: ElementEvents;
+
+  @bindable()
+  public theme: UxSelectTheme;
+
+  @bindable()
+  public autofocus: boolean | string;
+
+  @bindable({ defaultValue: false })
+  public disabled: boolean | string;
+
+  @bindable({ defaultValue: false })
+  public multiple: boolean | string;
+
+  @bindable public label: string;
+  @bindable public placeholder: string;
+
+  @bindable()
+  public variant: InputVariant = 'filled';
+
+  @bindable public dense: any = false;
+
+  public value: any;
+  public displayValue: string;
+  public expanded: boolean;
+
+  // Populated by Aurelia
+  public readonly optionWrapperEl: HTMLElement;
+  public readonly optionCtEl: UxOptionContainer;
+
+  constructor(
+    public readonly element: UxSelectElement,
+    private styleEngine: StyleEngine,
+    private observerLocator: ObserverLocator,
+    private taskQueue: TaskQueue,
+    private defaultConfiguration: UxDefaultSelectConfiguration
+  ) {
+    // Only chrome persist the element prototype when cloning with clone node
+    Object.setPrototypeOf(element, UxSelectElementProto);
+    if (this.defaultConfiguration.theme !== undefined) {
```

Comment:
@ben-girardet just for your awareness, we now have default configuration as well.